### PR TITLE
[ASTGen] Generate closure signatures

### PR DESCRIPTION
**Pull request by rintaro**
Still TODO:
* Capture list
* Anonymous parameters

### CODE REVIEW

File: test/ASTGen/decls.swift [hunk 1]
```diff
@@ -9,12 +9,11 @@
 
 // RUN: %diff -u %t/astgen.ast %t/cpp-parser.ast
 
-// RUN: %target-run-simple-swift(-Xfrontend -disable-availability-checking -enable-experimental-feature SwiftParser -enable-experimental-feature ParserASTGen -enable-experimental-feature ValueGenerics)
+// RUN: %target-run-simple-swift(-Xfrontend -disable-availability-checking -enable-experimental-feature ParserASTGen -enable-experimental-feature ValueGenerics)
 
 // REQUIRES: executable_test
 // REQUIRES: swift_swift_parser
 // REQUIRES: swift_feature_ParserASTGen
-// REQUIRES: swift_feature_SwiftParser
```
Comment (rintaro) [inline test/ASTGen/decls.swift:17]: These tests were unexpectedly disabled. `SwiftParser` doesn't exist (anymore?) but the command still had `-enable-experimental-feature SwiftParser`. cc: @drodriguez

**Comment by rintaro** [comment]
@swift-ci Please smoke test

File: test/ASTGen/decls.swift (hunk 1 above)
Comment (drodriguez) [reply test/ASTGen/decls.swift:17]: I will have a look at what can be done.

File: include/swift/AST/ASTBridging.h [hunk 2]
```diff
@@ -1230,10 +1250,28 @@ BridgedCallExpr BridgedCallExpr_createParsed(BridgedASTContext cContext,
...
+           "bracketRange:capturedSelfDecl:parameterList:asyncLoc:throwsLoc:"
+           "thrownType:arrowLoc:explicitResultType:inLoc:)")
 BridgedClosureExpr BridgedClosureExpr_createParsed(
     BridgedASTContext cContext, BridgedDeclContext cDeclContext,
-    BridgedParameterList cParamList, BridgedBraceStmt body);
+    BridgedDeclAttributes cAttributes, BridgedSourceRange cBracketRange,
+    BridgedNullableVarDecl cCapturedSelfDecl,
+    BridgedNullableParameterList cParameterList, BridgedSourceLoc cAsyncLoc,
+    BridgedSourceLoc cThrowsLoc, BridgedNullableTypeRepr cThrownType,
+    BridgedSourceLoc cArrowLoc, BridgedNullableTypeRepr cExcpliciResultType,
```
Comment (hamishknight) [inline include/swift/AST/ASTBridging.h:1265]:
```suggestion
    BridgedSourceLoc cArrowLoc, BridgedNullableTypeRepr cExplicitResultType,
```

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Review by hamishknight — APPROVED** [review]

**Comment by rintaro** [comment]
@swift-ci Please smoke test

File: test/ASTGen/decls.swift (hunk 1 above)
Comment (drodriguez) [reply test/ASTGen/decls.swift:17]: #77542

Found this one and a couple more. Seems that `SwiftParser` was renamed to `ParserASTGen`, but the old spelling was kept in the tests (together with the new spelling).

**Pull request merged**